java/core/src/main/java/co/worklytics/psoxy/gateway/HttpEventRequestDto.java

```diff
@@ -0,0 +1,88 @@
+package co.worklytics.psoxy.gateway;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+
+/**
+ * a JSON-serializable representation of an HttpEventRequest
+ */
+@Builder
+@Getter
+@Setter
+@NoArgsConstructor // for Jackson
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class HttpEventRequestDto implements HttpEventRequest {
+
+    private String path;
+    @JsonProperty("query")
+    private String query;
+    private String httpMethod;
+    private byte[] body;
+
+    @JsonProperty("clientIp")
+    private String clientIp;
+
+    private Map<String, List<String>> headers;
+    private Boolean https;
+
+    @JsonIgnore
+    @Override
+    public Optional<String> getClientIp() {
+        return Optional.ofNullable(clientIp);
+    }
+
+    @JsonIgnore
+    @Override
+    public Optional<String> getQuery() {
+        return Optional.ofNullable(this.query);
+    }
+
+    @Override
+    public Optional<String> getHeader(@NonNull String headerName) {
+        return Optional.ofNullable(headers.get(headerName))
+            .flatMap(list -> list.stream().findFirst());
+    }
+
+    @Override
+    public Optional<List<String>> getMultiValueHeader(@NonNull String headerName) {
+        return Optional.ofNullable(headers.get(headerName));
+    }
+
+    @JsonIgnore
+    @Override
+    public Object getUnderlyingRepresentation() {
+        throw new UnsupportedOperationException("Unimplemented method 'getUnderlyingRepresentation'");
+    }
+
+    @Override
+    public Optional<Boolean> isHttps() {
```

Comment:
why Boolean, can it be null ?? if this is an http event, it will either be https or not. 
Don't get why handle null here

Reply:
i don't recall. i think perhaps when triggered via one of the AWS gateway pathway, not necessarily known??